resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -2810,4 +2818,13 @@ protected void addResourceClassProcessor(ResourceClassProcessor processor)
    public ResourceBuilder getResourceBuilder() {
       return resourceBuilder;
    }
+
+   public static class ResteasyProviderFactoryException extends RuntimeException {
+
+      public ResteasyProviderFactoryException(String message) {

```

Comment:
Do we really need this additional class? The ResteasyProviderFactory is already quite big and critical memory-wise, maybe we can still use a RuntimeException, but with the i18n message ?